Can not use same file for `etcdKeyFile` & `etcdCertFile`

## Version

`1.0.0`

## Example `/etc/calico/calicoctl.yml` file

```
apiVersion: v1
kind: calicoApiConfig
metadata:
spec:
  datastoreType: etcdv2
  etcdEndpoints: https://etcd1:2379,https://etcd2:2379,https://etcd3:2379
  etcdKeyFile: /etc/tls/etcdclientpem
  etcdCertFile: /etc/tls/etcdclientpem
  etcdCACertFile: /etc/tls/etcdca
```
Where `/etc/tls/etcdclientpem` contains both tls `key` and `cert`.

## Command to reproduce
`sudo calicoctl node run --config /etc/calico/calicoctl.yml`

## Actual docker command triggered

`docker run --net=host --privileged --name=calico-node -d --restart=always -e ETCD_AUTHORITY= -e ETCD_CA_CERT_FILE=/etc/calico/certs/ca_cert.crt -e CALICO_NETWORKING_BACKEND=bird -e CALICO_LIBNETWORK_ENABLED=true -e CALICO_LIBNETWORK_IFPREFIX=cali -e ETCD_SCHEME= -e ETCD_KEY_FILE=/etc/calico/certs/key.pem -e ETCD_CERT_FILE=/etc/calico/certs/cert.crt -e NODENAME=swarm-node-0 -e NO_DEFAULT_POOLS= -e ETCD_ENDPOINTS=https://etcd1:2379,https://etcd2:2379,https://etcd3:2379 -v /etc/tls/etcdclientpem:/etc/calico/certs/cert.crt -v /var/run/calico:/var/run/calico -v /lib/modules:/lib/modules -v /var/log/calico:/var/log/calico -v /run/docker/plugins:/run/docker/plugins -v /var/run/docker.sock:/var/run/docker.sock -v /etc/tls/etcdca:/etc/calico/certs/ca_cert.crt calico/node:v1.0.0`

## Expected docker command

`docker run --net=host --privileged --name=calico-node -d --restart=always -e NO_DEFAULT_POOLS= -e CALICO_LIBNETWORK_ENABLED=true -e CALICO_LIBNETWORK_IFPREFIX=cali -e ETCD_ENDPOINTS=https://etcd1:2379,https://etcd2:2379,https://etcd3:2379 -e ETCD_AUTHORITY= -e ETCD_CA_CERT_FILE=/etc/calico/certs/ca_cert.crt -e NODENAME=swarm-node-0 -e CALICO_NETWORKING_BACKEND=bird -e ETCD_CERT_FILE=/etc/calico/certs/cert.crt -e ETCD_SCHEME= -e ETCD_KEY_FILE=/etc/calico/certs/key.pem -v /etc/tls/etcdca:/etc/calico/certs/ca_cert.crt -v /etc/tls/etcdclientpem:/etc/calico/certs/key.pem -v /etc/tls/etcdclientpem:/etc/calico/certs/cert.crt -v /var/run/calico:/var/run/calico -v /lib/modules:/lib/modules -v /var/log/calico:/var/log/calico -v /run/docker/plugins:/run/docker/plugins -v /var/run/docker.sock:/var/run/docker.sock calico/node:v1.0.0`

What we missed is: `-v /etc/tls/etcdclientpem:/etc/calico/certs/key.pem`

## Error log
```
goroutine 1 [running]:
panic(0x12f08c0, 0xc4203fb1b0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
main.main()
	/go/src/github.com/projectcalico/calico-containers/calico_node/startup.go:42 +0x50b
Calico node failed to start
time="2016-12-30T08:21:54Z" level=info msg="Creating Calico client" 
time="2016-12-30T08:21:54Z" level=info msg="Loading config from environment" 
panic: Error creating client: open /etc/calico/certs/key.pem: no such file or directory
```

--- a/calicoctl/commands/node/run.go
+++ b/calicoctl/commands/node/run.go
@@ -229,17 +229,23 @@ Description:
 		envs["IP6"] = ipv6
 	}
 
-	// Create a map of read only bindings.
-	vols := map[string]string{
-		logDir:            "/var/log/calico",
-		"/var/run/calico": "/var/run/calico",
-		"/lib/modules":    "/lib/modules",
+	// Create a struct for volumes to mount.
+	type vol struct {
+		hostPath      string
+		containerPath string
+	}
+
+	// vols is a slice of read only volume bindings.
+	vols := []vol{
+		{hostPath: logDir, containerPath: "/var/log/calico"},
+		{hostPath: "/var/run/calico", containerPath: "/var/run/calico"},
+		{hostPath: "/lib/modules", containerPath: "/lib/modules"},
 	}
 
 	if !disableDockerNw {
 		log.Info("Include docker networking volume mounts")
-		vols["/run/docker/plugins"] = "/run/docker/plugins"
-		vols["/var/run/docker.sock"] = "/var/run/docker.sock"
+		vols = append(vols, vol{hostPath: "/run/docker/plugins", containerPath: "/run/docker/plugins"},
+			vol{hostPath: "/var/run/docker.sock", containerPath: "/var/run/docker.sock"})
 	}
 
 	if etcdcfg.EtcdEndpoints == "" {
@@ -253,13 +259,14 @@ Description:
 	}
 	if etcdcfg.EtcdCACertFile != "" {
 		envs["ETCD_CA_CERT_FILE"] = ETCD_CA_CERT_NODE_FILE
-		vols[etcdcfg.EtcdCACertFile] = ETCD_CA_CERT_NODE_FILE
+		vols = append(vols, vol{hostPath: etcdcfg.EtcdCACertFile, containerPath: ETCD_CA_CERT_NODE_FILE})
+
 	}
 	if etcdcfg.EtcdKeyFile != "" && etcdcfg.EtcdCertFile != "" {
 		envs["ETCD_KEY_FILE"] = ETCD_KEY_NODE_FILE
-		vols[etcdcfg.EtcdKeyFile] = ETCD_KEY_NODE_FILE
+		vols = append(vols, vol{hostPath: etcdcfg.EtcdKeyFile, containerPath: ETCD_KEY_NODE_FILE})
 		envs["ETCD_CERT_FILE"] = ETCD_CERT_NODE_FILE
-		vols[etcdcfg.EtcdCertFile] = ETCD_CERT_NODE_FILE
+		vols = append(vols, vol{hostPath: etcdcfg.EtcdCertFile, containerPath: ETCD_CERT_NODE_FILE})
 	}
 
 	// Create the Docker command to execute (or display).  Start with the
@@ -282,8 +289,8 @@ Description:
 	}
 
 	// Add the volume mounts.
-	for k, v := range vols {
-		cmd = append(cmd, "-v", fmt.Sprintf("%s:%s", k, v))
+	for _, v := range vols {
+		cmd = append(cmd, "-v", fmt.Sprintf("%s:%s", v.hostPath, v.containerPath))
 	}
 
 	// Add the container image name

